### PR TITLE
Micronaut to 4.4.2 and Upgrades

### PR DESCRIPTION
**Pull request by timyates**
- Micronaut to 4.4.2
- Micronaut Gradle plugin to latest
- Unpin micronaut-data-mongodb
- Update Gradle to 8.7
- Switch to use `nativeCompile` instead of `nativeBuild`
- Switch base image for graal containers

@sdelamo @dstepanov

### CODE REVIEW

**Review by dstepanov — APPROVED** [review]

**Review by sdelamo — APPROVED** [review]

File: frameworks/Java/micronaut/gradle.properties [hunk 1]
```diff
@@ -1 +1 @@
-micronautVersion = 4.2.2
+micronautVersion=4.3.4
```
Comment (sdelamo) [inline frameworks/Java/micronaut/gradle.properties:1]:
```suggestion
micronautVersion=4.3.5
```

**Review by sdelamo — APPROVED** [review]

File: frameworks/Java/micronaut/buildSrc/build.gradle [hunk 2]
```diff
@@ -8,6 +8,6 @@ repositories {
 }
 
 dependencies {
-    implementation "io.micronaut.gradle:micronaut-gradle-plugin:4.0.1"
-    implementation "gradle.plugin.com.github.johnrengelman:shadow:7.1.2"
+    implementation "io.micronaut.gradle:micronaut-gradle-plugin:4.3.3"
```
Comment (sdelamo) [inline frameworks/Java/micronaut/buildSrc/build.gradle:11]:
```suggestion
    implementation "io.micronaut.gradle:micronaut-gradle-plugin:4.3.5"
```

File: frameworks/Java/micronaut/gradle.properties [hunk 3]
```diff
@@ -1 +1 @@
-micronautVersion = 4.2.2
+micronautVersion=4.3.5
```
Comment (sdelamo) [inline frameworks/Java/micronaut/gradle.properties:1]:
```suggestion
micronautVersion=4.3.7
```

**Review by sdelamo — CHANGES_REQUESTED** [review]

**Comment by timyates** [comment]
@dstepanov I think https://github.com/TechEmpower/FrameworkBenchmarks/pull/8778/commits/2df3b66dabca568d739a26a97714afba1a2885b5 is the right solution [for the failure](https://github.com/TechEmpower/FrameworkBenchmarks/actions/runs/8392954247/job/23062492002)?

**Review by sdelamo — APPROVED** [review]

**Comment by timyates** [comment]
@NateBrady23 Hi Nate!  Can you trigger the workflow? Hopefully this time will be green 🤞

File: frameworks/Java/micronaut/gradle.properties [hunk 4]
```diff
@@ -1 +1 @@
-micronautVersion = 4.2.2
+micronautVersion=4.4.1
```
Comment (sdelamo) [inline frameworks/Java/micronaut/gradle.properties:1]:
```suggestion
micronautVersion=4.4.2
```

**Review by sdelamo — CHANGES_REQUESTED** [review]

**Comment by timyates** [comment]
@NateBrady23 Sorry about this... It looks like the failure was an [issue withe the mongo tests](https://github.com/TechEmpower/FrameworkBenchmarks/actions/runs/8949311129/job/24649004953#step:9:1051) not being able to find `tf-database` on the docker network 🤔 But as far as I can tell, nothing has changed here...

Could it be a temporary docker issue that's caused the failure?

**Comment by NateBrady23** [comment]
I think the issue was on our end, but should be fixed in the main branch now. Thank you!

**Pull request merged**